### PR TITLE
fix: update variables for latest version of key protect

### PR DESCRIPTION
**Pull request by MatthewLemmond**
### Description
Update the variables in the module for use with the latest version of terraform-ibm-key-protect-all-inclusive

**Check the relevant boxes:**
- [ ] Bug fix (nonbreaking change that fixes an issue)
- [ ] New feature (nonbreaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Examples or tests (addition or updates of examples or tests)
- [ ] Documentation update
- [ ] CI related update (pipeline, etc.)

### Checklist

- [ ] If relevant, a test for the change has been added or updated as part of this PR.
- [ ] If relevant, documentation for the change has been added or updated as part of this PR.

### Merge

- Merge using "Squash and merge".
- Make sure to use a relevant [conventional commit](https://www.conventionalcommits.org/) message that is based on the PR contents. The commit message determines whether a new version of the modules needs to be released, and if so, which semver number to use).

### CODE REVIEW

**Review by ocofaigh — CHANGES_REQUESTED** [review]
@MatthewLemmond We need to also add a variable called `existing_cos_instance_guid` for the exact same reason.

File: variables.tf [hunk 1]
```diff
@@ -55,6 +55,12 @@ variable "cos_instance_name" {
   default     = null
 }
 
+variable "existing_cos_instance_id" {
+  description = "ID of an existing cos instance where the bucket will be created"
```
Comment (ocofaigh) [inline variables.tf:59]: Lets be consistent..

The GUID of an existing COS instance, required if 'var.create_cos_instance' is false.

File: variables.tf [hunk 2]
```diff
@@ -138,8 +144,14 @@ variable "create_key_protect_instance" {
...
 
+variable "existing_key_protect_instance_guid" {
+  description = "The GUID of an existing Key Protect instance, required if 'var.create_key_protect_instance' is false."
+  type        = string
+  default     = null
+}
+
 variable "key_protect_instance_name" {
-  description = "Name to set as the instance name if creating a Key Protect instance, otherwise name of an existing Key Protect instance to use, this instance will store the Key used to encrypt the data in the COS Bucket"
+  description = "The name to give the Key Protect instance that will be provisioned by this module, this variable will be ignored if a value is passed for 'var.existing_key_protect_instance_guid."
```
Comment (ocofaigh) [inline variables.tf:154]: Missing a closing single quote

File: variables.tf (hunk 1 above)
Comment (ocofaigh) [reply variables.tf:59]: Also update the description for var.cos_instance_name to be consistent too...

The name to give the COS instance that will be provisioned by this module. This variable will be ignored if a value is passed for 'var.existing_cos_instance_id'.

File: main.tf [hunk 3]
```diff
@@ -13,11 +13,11 @@ locals {
...
 
-  #ensure if create_cos_instance = false, then cos_instance_name is provided
-  cos_validate_condition = (!var.create_cos_instance && var.cos_instance_name == null)
-  cos_validate_msg       = "If create_cos_instance is false, then provide the cos_instance_name to create buckets"
+  #ensure if create_cos_instance = false, then existing_cos_instance_id is provided
+  cos_id_validate_condition = (!var.create_cos_instance && var.existing_cos_instance_id == null)
+  cos_id_validate_msg       = "If create_cos_instance is false, then provide the existing_cos_instance_id to create buckets"
   # tflint-ignore: terraform_unused_declarations
-  cos_validate_check = regex("^${local.cos_validate_msg}$", (!local.cos_validate_condition ? local.cos_validate_msg : ""))
+  cos_id_validate_check = regex("^${local.cos_id_validate_msg}$", (!local.cos_id_validate_condition ? local.cos_id_validate_msg : ""))
```
Comment (ocofaigh) [inline main.tf:20]: ^^ Are we missing the same logic check for `existing_key_protect_instance_guid` and `create_key_protect_instance` ?

File: main.tf [hunk 4]
```diff
@@ -64,25 +65,16 @@ resource "ibm_resource_instance" "cos_instance" {
 }
 
 locals {
-  cos_instance_id      = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].id)[0] : tolist(data.ibm_resource_instance.cos_instance[*].id)[0]
-  cos_instance_guid    = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].guid)[0] : tolist(data.ibm_resource_instance.cos_instance[*].guid)[0]
+  cos_instance_id      = var.create_cos_instance == true ? tolist(ibm_resource_instance.cos_instance[*].id)[0] : var.existing_cos_instance_id
+  cos_instance_name    = var.cos_instance_name == null ? "${var.environment_name}-cos" : var.cos_instance_name
```
Comment (ocofaigh) [inline main.tf:69]: Can we update the variable for `var.cos_instance_name` to ensure it says if null is passed, and `var.create_cos_instance` is true, that the new COS instance name will default to `${var.environment_name}-cos`

**Review by ocofaigh — CHANGES_REQUESTED** [review]
Left some comments. Also remove common-dev-assets from the PR. You committed an old version of it. No need to include it in the PR

File: main.tf (hunk 3 above)
Comment (MatthewLemmond) [reply main.tf:20]: This logic exists for Key Protect at the Key Protect All Inclusive level [here](https://github.com/terraform-ibm-modules/terraform-ibm-key-protect-all-inclusive/blob/main/main.tf#L6).

**Comment by ocofaigh** [comment]
@MatthewLemmond I have taken the code from your branch and included it as part of the refactoring PR https://github.com/terraform-ibm-modules/terraform-ibm-cos/pull/99 so going to close this PR. Please check out that PR

**Pull request closed without merging**